internal/test/func-e.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// fakeFuncESrcPath is a path to the test source file used to simulate func-e which runs envoy as its child.
+	fakeFuncESrcPath = filepath.Join(funcEGoModDir, "internal", "test", "testdata", "fake_func-e", "fake_func-e.go")
```

Comment:
done in the other branch, but noting for completion that go:embed is superior to this style if for no other reason IDEs can help make it work (ex refactoring and navigation). OTOH you have to play by the rules and only do it where the path you need is enclosed by the current directory downwards.

another nit you'll notice in the other change: we aren't ambiguous in the testdata about the main package, so we can use fake_func-e.go instead of making a directory and then a file named almost the same as the directory. If later the moreos testdata gets crowded, since we are using go:embed at least renaming or moving that with goland will fix the go:embed reference. IOTW we can push any tension about directory contents (ex don't need to pre-emptively make a dir with only one file in it), because refactoring later is easy with go:embed.